From 01386018108fbf1b1b1b334754f9a7b62004b371 Mon Sep 17 00:00:00 2001
From: "Friedrich W. H. Kossebau" <kossebau@kde.org>
Date: Wed, 23 Feb 2022 20:43:08 +0100
Subject: [PATCH] Fixup last commits

* asturm 2025-06-22: Extract parts that switched to CMakePackageConfigHelpers
  and improved other ECM module includes (ordering, use KDECompilerSettings
  instead of KDEFrameworkCompilerSettings)

---
 CMakeLists.txt                   | 9 ++++-----
 src/KChart/CMakeLists.txt        | 3 ++-
 src/KGantt/CMakeLists.txt        | 3 ++-
 3 files changed, 8 insertions(+), 7 deletions(-)

diff --git a/CMakeLists.txt b/CMakeLists.txt
index 35aefd20..80c7b861 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -8,21 +8,20 @@
 set(CMAKE_MODULE_PATH ${ECM_MODULE_PATH} ${ECM_KDE_MODULE_DIR})
 
 include(KDEInstallDirs)
-include(KDEFrameworkCompilerSettings NO_POLICY_SCOPE)
 include(KDECMakeSettings)
-
-include(FeatureSummary)
-include(GenerateExportHeader)
+include(KDECompilerSettings NO_POLICY_SCOPE)
 
 include(ECMGenerateHeaders)
 include(ECMSetupVersion)
-include(ECMPackageConfigHelpers)
 include(ECMGeneratePriFile)
 include(ECMPoQmTools)
 include(ECMAddTests)
 set(QT_REQUIRED_VERSION "5.12.0")
 
+include(CMakePackageConfigHelpers)
 include(CMakeDependentOption)
+include(FeatureSummary)
+include(GenerateExportHeader)
 
 set(HAS_ECMAddQch TRUE)
 include(ECMAddQch)
diff --git a/src/KChart/CMakeLists.txt b/src/KChart/CMakeLists.txt
index f7adf0b0..39380806 100644
--- a/src/KChart/CMakeLists.txt
+++ b/src/KChart/CMakeLists.txt
@@ -340,7 +340,7 @@ if (BUILD_QCH)
     set(PACKAGE_INCLUDE_QCHTARGETS "include(\"\${CMAKE_CURRENT_LIST_DIR}/KChartQchTargets.cmake\")")
 endif()
 
-ecm_configure_package_config_file(
+configure_package_config_file(
     "${CMAKE_CURRENT_SOURCE_DIR}/KChartConfig.cmake.in"
     "${CMAKE_CURRENT_BINARY_DIR}/KChartConfig.cmake"
     INSTALL_DESTINATION ${CMAKECONFIG_INSTALL_DIR}
@@ -357,6 +357,7 @@ install(EXPORT KChartTargets
         FILE KChartTargets.cmake)
 
 ecm_generate_pri_file(
+    VERSION ${KCHARTLIB_VERSION}
     BASE_NAME KChart
     LIB_NAME KChart
     DEPS "widgets svg"
diff --git a/src/KGantt/CMakeLists.txt b/src/KGantt/CMakeLists.txt
--- a/src/KGantt/CMakeLists.txt
+++ b/src/KGantt/CMakeLists.txt
@@ -164,7 +164,7 @@ if (BUILD_QCH)
     set(PACKAGE_INCLUDE_QCHTARGETS "include(\"\${CMAKE_CURRENT_LIST_DIR}/KGanttQchTargets.cmake\")")
 endif()
 
-ecm_configure_package_config_file(
+configure_package_config_file(
     "${CMAKE_CURRENT_SOURCE_DIR}/KGanttConfig.cmake.in"
     "${CMAKE_CURRENT_BINARY_DIR}/KGanttConfig.cmake"
     INSTALL_DESTINATION ${CMAKECONFIG_INSTALL_DIR}
@@ -181,6 +181,7 @@ install(EXPORT KGanttTargets
         FILE KGanttTargets.cmake)
 
 ecm_generate_pri_file(
+    VERSION ${KGANTTLIB_VERSION}
     BASE_NAME KGantt
     LIB_NAME KGantt
     DEPS "widgets printsupport"
-- 
GitLab
